internal/ingress/annotations/conditions/types.go

```diff
@@ -0,0 +1,255 @@
+	//
+	// You can't use an HTTP header condition to specify the host header. Use HostHeaderConditionConfig
+	// to specify a host header condition.
+	HttpHeaderName *string
+
+	// One or more strings to compare against the value of the HTTP header. The
+	// maximum size of each string is 128 characters. The comparison strings are
+	// case insensitive. The following wildcard characters are supported: * (matches
+	// 0 or more characters) and ? (matches exactly 1 character).
+	//
+	// If the same header appears multiple times in the request, we search them
+	// in order until a match is found.
+	//
+	// If you specify multiple strings, the condition is satisfied if one of the
+	// strings matches the value of the HTTP header. To require that all of the
+	// strings are a match, create one condition per string.
+	Values []*string
+}
+
+func (c *HttpHeaderConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about an HTTP method condition.
+//
+// HTTP defines a set of request methods, also referred to as HTTP verbs. For
+// more information, see the HTTP Method Registry (https://www.iana.org/assignments/http-methods/http-methods.xhtml).
+// You can also define custom HTTP methods.
+type HttpRequestMethodConditionConfig struct {
+	// The name of the request method. The maximum size is 40 characters. The allowed
+	// characters are A-Z, hyphen (-), and underscore (_). The comparison is case
+	// sensitive. Wildcards are not supported; therefore, the method name must be
+	// an exact match.
+	//
+	// If you specify multiple strings, the condition is satisfied if one of the
+	// strings matches the HTTP request method. We recommend that you route GET
+	// and HEAD requests in the same way, because the response to a HEAD request
+	// may be cached.
+	Values []*string
+}
+
+func (c *HttpRequestMethodConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about a path pattern condition.
+type PathPatternConditionConfig struct {
+	// One or more path patterns to compare against the request URL. The maximum
+	// size of each string is 128 characters. The comparison is case sensitive.
+	// The following wildcard characters are supported: * (matches 0 or more characters)
+	// and ? (matches exactly 1 character).
+	//
+	// If you specify multiple strings, the condition is satisfied if one of them
+	// matches the request URL. The path pattern is compared only to the path of
+	// the URL, not to its query string. To compare against the query string, use
+	// QueryStringConditionConfig.
+	Values []*string
+}
+
+func (c *PathPatternConditionConfig) validate() error {
+	if len(c.Values) == 0 {
+		return errors.New("Values cannot be empty")
+	}
+	return nil
+}
+
+// Information about a key/value pair.
+type QueryStringKeyValuePair struct {
+	// The key. You can omit the key.
```

Comment:
What does it mean if you omit the key?
`// The key, optional. If omitted, XXXX???`

Reply:
it should be any param equal to valueXXX. (however, my tests didn't seems so, looks like an bug in ALB side).
The comments is copied directly from awsSDKGO :D